redbot/cogs/admin/admin.py

```diff
@@ -450,45 +450,70 @@ async def selfroleset(self, ctx: commands.Context):
         pass
 
     @selfroleset.command(name="add")
-    async def selfroleset_add(self, ctx: commands.Context, *, role: discord.Role):
+    async def selfroleset_add(self, ctx: commands.Context, *roles: discord.Role):
         """
-        Add a role to the list of available selfroles.
+        Add a role, or a selection of roles, to the list of available selfroles.
 
         NOTE: The role is case sensitive!
         """
-        if not self.pass_user_hierarchy_check(ctx, role):
-            await ctx.send(
-                _(
-                    "I cannot let you add {role.name} as a selfrole because that role is higher than or equal to your highest role in the Discord hierarchy."
-                ).format(role=role)
-            )
-            return
-        async with self.config.guild(ctx.guild).selfroles() as curr_selfroles:
-            if role.id not in curr_selfroles:
-                curr_selfroles.append(role.id)
-                await ctx.send(_("Added."))
+        count = 0
+        current_selfroles = await self.config.guild(ctx.guild).selfroles()
+        for role in roles:
+            if not self.pass_user_hierarchy_check(ctx, role):
+                await ctx.send(
+                    _(
+                        "I cannot let you add {role.name} as a selfrole because that role is higher than or equal to your highest role in the Discord hierarchy."
```

Comment:
Styling
```suggestion
                        "I cannot let you add {role.name} as a selfrole because that role is"
                        " higher than or equal to your highest role in the Discord hierarchy."
```